Source/stash.cpp

```diff
@@ -0,0 +1,618 @@
+				if (itemId == -it)
+					itemId = 0;
+			}
+		}
+	}
+
+	AddItemToStashGrid(firstSlot, it, itemSize);
+
+	if (cn == CURSOR_HAND && !IsHardwareCursor())
+		SetCursorPos(MousePosition + Displacement(cursSize / 2));
+	NewCursor(cn);
+}
+
+void CheckStashCut(Point cursorPosition, bool automaticMove, bool dropItem)
+{
+	auto &player = Players[MyPlayerId];
+
+	if (withdrawGoldFlag) {
+		withdrawGoldFlag = false;
+		withdrawGoldValue = 0;
+	}
+
+	Point slot = { -1, -1 };
+
+	for (int y = 0; y < 10; y++) {
+		for (int x = 0; x < 10; x++) {
+			Point testSlot = { x, y };
+			Rectangle cell {
+				GetPanelPosition(UiPanels::Stash, StashCellPosition(testSlot)),
+				{ InventorySlotSizeInPixels.width + 1, InventorySlotSizeInPixels.height + 1 }
+			};
+
+			// check which inventory rectangle the mouse is in, if any
+			if (cell.Contains(cursorPosition)) {
+				slot = testSlot;
+				break;
+			}
+		}
+	}
+
+	if (slot == Point { -1, -1 }) {
+		return;
+	}
+
+	Item &holdItem = player.HoldItem;
+	holdItem._itype = ItemType::None;
+
+	bool automaticallyMoved = false;
+	bool automaticallyEquipped = false;
+
+	int16_t ii = Stash.stashGrids[Stash.page][slot.x][slot.y];
+	if (ii != 0) {
+		int iv = abs(ii) - 1;
+
+		holdItem = Stash.stashList[iv - 1];
+		if (automaticMove) {
+			if (CanBePlacedOnBelt(holdItem)) {
+				automaticallyMoved = AutoPlaceItemInBelt(player, holdItem, true);
+			} else {
+				automaticallyMoved = automaticallyEquipped = AutoEquip(MyPlayerId, holdItem);
+			}
+		}
+
+		if (!automaticMove || automaticallyMoved) {
+			Stash.RemoveStashItem(iv - 1);
+		}
+	}
+
+	if (!holdItem.isEmpty()) {
+		CalcPlrInv(player, true);
+		CheckItemStats(player);
+		if (automaticallyEquipped) {
+			PlaySFX(ItemInvSnds[ItemCAnimTbl[holdItem._iCurs]]);
+		} else if (!automaticMove || automaticallyMoved) {
+			PlaySFX(IS_IGRAB);
+		}
+
+		if (automaticMove) {
+			if (!automaticallyMoved) {
+				if (CanBePlacedOnBelt(holdItem)) {
+					player.SaySpecific(HeroSpeech::IHaveNoRoom);
+				} else {
+					player.SaySpecific(HeroSpeech::ICantDoThat);
+				}
+			}
+
+			holdItem._itype = ItemType::None;
+		} else {
+			NewCursor(holdItem._iCurs + CURSOR_FIRSTITEM);
+			if (!IsHardwareCursor() && !dropItem) {
+				// For a hardware cursor, we set the "hot point" to the center of the item instead.
+				SetCursorPos(cursorPosition - Displacement(cursSize / 2));
+			}
+		}
+	}
+
+	if (dropItem) {
+		TryDropItem();
+	}
+}
+
+void StartGoldWithdraw()
+{
+	initialWithdrawGoldValue = Stash.gold;
+
+	if (talkflag)
+		control_reset_talk();
+
+	Point start = GetPanelPosition(UiPanels::Stash, { 67, 128 });
+	SDL_Rect rect = MakeSdlRect(start.x, start.y, 180, 20);
+	SDL_SetTextInputRect(&rect);
+
+	withdrawGoldFlag = true;
+	withdrawGoldValue = 0;
+	SDL_StartTextInput();
+}
+
+} // namespace
+
+constexpr Point StashCellPosition(Point slot)
+{
+	constexpr int StashNextCell = INV_SLOT_SIZE_PX + 1; // spacing between each cell
+
+	return slot * StashNextCell + Displacement { 17, 76 };

```

Comment:
```suggestion
	return slot * StashNextCell + Displacement { 17, 47 };
```
This displacement causes the top row of cells to be inaccessible, 47 is `76 - StashNextCell` but might not be correct either.

Reply:
76 was assuming the position relevant for the draw function rather then the logical one.